homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,215 @@
+                        default=int(self.discovery_info.get(CONF_PORT, DEFAULT_PORT)),
+                    ): int,
+                    vol.Optional(CONF_SSL, default=DEFAULT_SSL): bool,
+                    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
+                    vol.Optional(CONF_TOKEN): str,
+                }
+            )
+            return self.async_show_form(step_id="manual_setup", data_schema=data_schema)
+
+        host = user_input.pop(CONF_HOST)
+        port = user_input.pop(CONF_PORT)
+        prefix = "https" if user_input.get(CONF_SSL) else "http"
+        user_input[CONF_URL] = f"{prefix}://{host}:{port}"
+        return await self.async_step_user(user_input=user_input)
+
+    async def async_step_select_server(self, user_input=None):
+        """Use selected Plex server."""
+        if user_input is None:
+            return await self.async_step_user()
+
+        config = self.current_login
+        config[CONF_SERVER] = user_input.get(CONF_SERVER)
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_discovery(self, discovery_info):
+        """Set default host and port from discovery."""
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        self.discovery_info = discovery_info
+        return await self.async_step_user()
+
+    async def async_step_import_plex_conf(self, import_config):
+        """Import from Plex media_player file config.
+
+        Legacy.
+        """
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        host_and_port, host_config = import_config.popitem()
+        prefix = "https" if host_config[CONF_SSL] else "http"
+        url = f"{prefix}://{host_and_port}"
+
+        config = {
+            CONF_URL: url,
+            CONF_TOKEN: host_config[CONF_TOKEN],
+            CONF_VERIFY_SSL: host_config["verify"],
+        }
+
+        _LOGGER.info("Imported configuration from legacy config file")
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_import(self, import_config):
+        """Import from Plex configuration."""
+        if self._async_in_progress():
+            return self.async_abort(reason="already_in_progress")
+
+        host = import_config.get(CONF_HOST)
+        port = import_config[CONF_PORT]
+        token = import_config.get(CONF_TOKEN)
+        server = import_config.get(CONF_SERVER)
+
+        if host and port:
+            prefix = "https" if import_config[CONF_SSL] else "http"
+            url = f"{prefix}://{host}:{port}"
+
+            config = {
+                CONF_URL: url,
+                CONF_TOKEN: token,
+                CONF_VERIFY_SSL: import_config[CONF_VERIFY_SSL],
+            }
+        elif token:
+            config = {CONF_TOKEN: token, CONF_SERVER: server}
+        else:
+            return self.async_abort(reason="invalid_import")
```

Comment:
When can this happen?

Reply:
When the config is supplied without a host or token.

Reply:
That should be impossible based on the config schema. Require at least one key.

Reply:
The server can be configured with one of these (minimized) combinations of parameters:
- token
- token + server
- host (no auth)
- host + token

But the user may want to add the configuration purely to set one of the the `media_player` options and configure the server via config flow:
```yaml
plex:
  media_player:
    use_episode_art: true
```
This would leave the server section blank but would pass along the defaults. This should _not_ attempt to create a config entry.

Reply:
We should not allow only configuring the media_player options in config yaml. Then the user have to configure the server too.

Please update the config schema so we can remove the check here.

Reply:
Done in #26583